pkg/new-ui/v1beta1/frontend/src/app/constants/algorithms-settings.const.ts

```diff
@@ -0,0 +1,208 @@
+import {
+  AlgorithmsEnum,
+  AlgorithmSettingType,
+} from '../enumerations/algorithms.enum';
+
+export interface AlgorithmSetting {
+  name: string;
+  value: any;
+  values?: any[];
+  type: AlgorithmSettingType;
+}
+
+export const GridSettings: AlgorithmSetting[] = [];
+
+export const RandomSearchSettings: AlgorithmSetting[] = [
+  {
+    name: 'random_state ',
+    value: 'None',
+    type: AlgorithmSettingType.STRING,
```

Comment:
We don't have default value for the random state, but it should be an integer value.
```suggestion
    type: AlgorithmSettingType.INTEGER,
```
Maybe we should add algorithm settings to the Experiment YAML only if user wants it ?
For example, add check box to the form.

It is Suggestion's CR responsibility to add default values for the settings.
In case of random state, it should be empty if user doesn't want it.

Reply:
I'd really like have the UI show all the options a user can edit.

So I'd propose to go the other way around it. 
* Expose all params with default values, I'm using the values I see in the docs
* For parameters like `random_state` where the default should be None [ which I guess means the param should not be present right? ] 
    * the UI will have an empty value by default
    * if the user doesn't provide a value the UI will completely omit this param

This way we keep the UX of showing to the user everything they can configure without breaking the default expected behavior. WDYT?

Reply:
> For parameters like random_state where the default should be None [ which I guess means the param should not be present right? ]

You are right, the value just should be omitted in the `AlgorithmSettings` list in the Experiment.

Otherwise, if we send `None` value in the `random_state` parameter, we get an error here: https://github.com/kubeflow/katib/blob/master/pkg/suggestion/v1beta1/skopt/service.py#L72-L73.  

> the UI will have an empty value by default
if the user doesn't provide a value the UI will completely omit this param

Sounds good, we need to make sure that in case of empty values frontend should not add `AlgorithmSettings` to the Experiment.

Reply:
Correct. So looking at the docs I only see that the `random_state` is the one that could be omitted.
Are there any other algorithm params with such a behavior currently?

Reply:
I believe `sigma` parameter in CMA-ES also can be omitted, is that correct @c-bata ?

Reply:
Yes! If omitted, `sigma` is automatically calculated from the search space definition like [this](https://github.com/c-bata/goptuna/blob/v0.8.0/cmaes/sampler.go#L296).

Furthermore, `ei_candidates` and `startup_trials` params for Goptuna's TPE algorithm can also be omitted.

https://github.com/kubeflow/katib/blob/fe64c18172ba370fb3934552c50442f1da21ec1a/pkg/suggestion/v1beta1/goptuna/converter.go#L51

https://github.com/kubeflow/katib/blob/fe64c18172ba370fb3934552c50442f1da21ec1a/pkg/suggestion/v1beta1/goptuna/converter.go#L57

Reply:
Great, so `random_state` and `sigma` for now.

> Furthermore, `ei_candidates` and `startup_trials` params for Goptuna's TPE algorithm can also be omitted.

@andreyvelich @c-bata looking at the docs I couldn't find any parameters for the TPE algorithm. Could you provide me an extensive list of the supported params for this algorithm?

Reply:
Thank you @c-bata!
@kimwnasptd For TPE we don't need to add goptuna settings to the UI since by default we use hyperopt Suggestion: https://github.com/kubeflow/katib/blob/master/manifests/v1beta1/katib-controller/katib-config.yaml#L39.

@kimwnasptd  If you want to add TPE setting to the UI this is the list: https://github.com/kubeflow/katib/blob/master/pkg/suggestion/v1beta1/hyperopt/service.py#L90-L101.

Reply:
Thank you @andreyvelich!

Do these params have default values or are they just like `random_state`, with default value of "None"?

Reply:
I believe they don't have default values, so let's keep them empty.

Reply:
Pushed two more commits that:
1. Make the UI not send an Algorithm parameter in case the user didn't provide a value
2. Added default parameters for TPE based on the provided params

here's also an example JSON body
![Screenshot from 2021-03-01 19-25-39](https://user-images.githubusercontent.com/11134742/109534938-96603c00-7ac4-11eb-89e2-b11f9b7d419e.png)